// SPDX-License-Identifier: Apache-2.0
// Copyright 2013-2023 The Foundry Visionmongers Ltd
syntax = "proto3";

package openassetio_grpc_proto;

message EmptyRequest {}
message EmptyResponse {}


// Factory

message IdentifiersResponse {
	repeated string identifer = 1;
};


message InstantiateRequest {
    string identifier = 1;
}

message InstantiateResponse {
    string handle = 1;
}


message DestroyRequest {
    string handle = 1;
}

// Manager Interface

message IdentifierRequest {
    string handle = 1;
}

message IdentifierResponse {
    string identifier = 1;
}

message DisplayNameRequest {
    string handle = 1;
}

message DisplayNameResponse {
    string displayName = 1;
}

// Service

service ManagerProxy  {
	rpc Identifiers(EmptyRequest) returns (IdentifiersResponse);
    rpc Instantiate(InstantiateRequest) returns (InstantiateResponse);
    rpc Destroy(DestroyRequest) returns (EmptyResponse);
    // ManagerInterface method
    rpc Identifier(IdentifierRequest) returns (IdentifierResponse);
    rpc DisplayName(DisplayNameRequest) returns (DisplayNameResponse);
};

